sdk/core/azure-core/src/main/java/com/azure/core/http/rest/RestProxy.java

```diff
@@ -365,10 +378,11 @@ private static Exception instantiateUnexpectedException(final UnexpectedExceptio
      * @return An async-version of the provided decodedResponse.
      */
     private Mono<HttpDecodedResponse> ensureExpectedStatus(final HttpDecodedResponse decodedResponse,
-        final SwaggerMethodParser methodParser) {
+        final SwaggerMethodParser methodParser, RequestOptions options) {
         final int responseStatusCode = decodedResponse.getSourceResponse().getStatusCode();
         final Mono<HttpDecodedResponse> asyncResult;
-        if (!methodParser.isExpectedResponseStatusCode(responseStatusCode)) {
+        if (!methodParser.isExpectedResponseStatusCode(responseStatusCode)
+                && (options == null || options.getStatusOption() != ResponseStatusOption.NO_THROW)) {
```

Comment:
Can we have this be `== ResponseStatusOption.THROW`, or whatever the rename becomes. The anti-case could begin failing if additional enum options are provided.

Reply:
changed to `isThrowOnError()`